Sources/Rendering/Core/RenderWindowInteractor/index.d.ts

```diff
@@ -113,398 +127,398 @@ export interface vtkRenderWindowInteractor extends vtkObject {
 
 	/**
 	 *
 	 */
-	invokeKeyPress(callData: object): any;
+	invokeKeyPress(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeKeyDown(callData: object): any;
+	invokeKeyDown(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeKeyUp(callData: object): any;
+	invokeKeyUp(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeStartMouseWheel(callData: object): any;
+	invokeStartMouseWheel(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeMouseWheel(callData: object): any;
+	invokeMouseWheel(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeEndMouseWheel(callData: object): any;
+	invokeEndMouseWheel(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeStartPinch(callData: object): any;
+	invokeStartPinch(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokePinch(callData: object): any;
+	invokePinch(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeEndPinch(callData: object): any;
+	invokeEndPinch(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeStartPan(callData: object): any;
+	invokeStartPan(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokePan(callData: object): any;
+	invokePan(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeEndPan(callData: object): any;
+	invokeEndPan(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeStartRotate(callData: object): any;
+	invokeStartRotate(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeRotate(callData: object): any;
+	invokeRotate(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeEndRotate(callData: object): any;
+	invokeEndRotate(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeButton3D(callData: object): any;
+	invokeButton3D(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeMove3D(callData: object): any;
+	invokeMove3D(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeStartPointerLock(callData: object): any;
+	invokeStartPointerLock(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeEndPointerLock(callData: object): any;
+	invokeEndPointerLock(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeStartInteractionEvent(callData: object): any;
+	invokeStartInteractionEvent(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeInteractionEvent(callData: object): any;
+	invokeInteractionEvent(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
 	 */
-	invokeEndInteractionEvent(callData: object): any;
+	invokeEndInteractionEvent(callData: IRenderWindowInteractorEvent): any;
 
 	/**
 	 *
-	 * @param args
+	 * @param cb The callback to be called
 	 */
-	onStartAnimation(args: any): any;
+	onStartAnimation(cb: InteractorEventCallback): Readonly<vtkSubscription>;
```

Comment:
The [on${_capitalize(eventName)}](https://github.com/Kitware/vtk-js/blob/master/Sources/macros.js#L943) does have two args `callback, priority = 0.0 => callback:InteractorEventCallback, priority?:number`, if i am not wrong the callback needs to have an [apply](https://github.com/Kitware/vtk-js/blob/master/Sources/macros.js#L944-L947) attribute otherwise the function will exit.

@jourdain can you give us your input on this?

Reply:
So the [apply](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply) is only to check that it is a `function`. It will be called with `this` resolving to the publicAPI of the event source where `invokeXXX` was called.

Reply:
But technically, it could also be an object with as apply function ;-).